subsys/net/lib/ftp_client/src/ftp_client.c

```diff
@@ -346,26 +348,31 @@ static void do_ftp_recv_data(const char *pasv_msg)
 
 static int poll_data_task_done(void)
 {
-	int ret = -ENODATA;
-	int wait_time = 0;
+	int ret = 0;
+	unsigned int wait_time_ms = 0;
 
-	do {
-		ret = do_ftp_recv_ctrl(true, FTP_CODE_226);
-		if (ret == FTP_CODE_226) {
+	while (ret >= 0 || ret == -ETIMEDOUT) {
```

Comment:
Just a note that this loop had a lot going on. I would have perhaps considered separating the data task running modification to it's own while loop, even if it would add some extra code, it would likely be easier to follow.